src/back/glsl/mod.rs

```diff
@@ -2015,6 +2015,30 @@ impl<'a, W: Write> Writer<'a, W> {
 
                 writeln!(self.out, "{}barrier();", level)?;
             }
+            Statement::WorkgroupUniformLoad { pointer, result } => {
+                writeln!(self.out, "{}memoryBarrierShared();", level)?;
+                writeln!(self.out, "{}barrier();", level)?;
```

Comment:
We recently introduced a new function `write_barrier` for all backends that we could use.